test/e2e/quadlet_test.go

```diff
@@ -116,6 +116,25 @@ func (t *quadletTestcase) assertKeyIs(args []string, unit *parser.UnitFile) bool
 	return true
 }
 
+func (t *quadletTestcase) assertKeyIsRegex(args []string, unit *parser.UnitFile) bool {
+	group := args[0]
+	key := args[1]
+	values := args[2:]
```

Comment:
Mini nit: length checks would make it more robust (during development of new test cases)